src-tauri/src/main.rs

```diff
@@ -136,6 +139,10 @@ fn on_tray_event(
         SystemTrayEvent::MenuItemClick { id, .. } => match id.as_str() {
             "quit" => {
                 println!("system tray received a quit click");
+                let mut manager_state = manager_state.lock().unwrap();
+                manager_state.alive = false;
+                drop(manager_state);
+                sleep(Duration::from_millis(500)); //TODO: revisit this
```

Comment:
I have a few ideas for getting rid of this... Having a variable that tracks number of running watchers. It's a bit tricky to implement as I have drop the mutex so that it's acquired by the start_watcher threads, Then reacquire it and wait till the count is zero before proceeding. Will try it tomorrow!